webapi/Lib/WebHooks/MccSoft.WebHooks/Manager/WebHookManager.cs

```diff
@@ -100,6 +115,25 @@ public async Task Unsubscribe(Guid subscriptionId)
         await localDbContext.SaveChangesAsync();
     }
 
+    /// <inheritdoc />
+    public async Task<string> RotateSecret(Guid subscriptionId)
+    {
+        var webHookSubscription = await _dbContext
+            .WebHookSubscriptions<TSub>()
+            .FirstAsync(webhook => webhook.Id == subscriptionId);
+
+        var secret = _signatureService.GenerateEncryptedSecret(webHookSubscription);
+        if (string.IsNullOrWhiteSpace(secret))
+            throw new InvalidOperationException(
+                $"{nameof(IWebHookOptionBuilder<TSub>.UseSigning)} is disabled or EncryptionKey is missing."
+            );
+
+        _dbContext.Update(webHookSubscription);
```

Comment:
why do we need this line? I guess, it will work without it, since we retrieved `webHookSubscription` from the same db context

Reply:
you’re right, will remove it

probably it is an artefact from previous idea - where used public method for update